torch/fx/graph.py

```diff
@@ -707,15 +722,37 @@ def emit_node(node : Node):
             emit_node(node)
             delete_unused_values(node)
 
-        # repr() for inf and nan floating point values aren't parseable by
-        # python as literals. Explicitly import the names from the ``math`` module.
-        import_strs: Set[str] = set()
+        # Handle imports
+        #
+        # First, demangle qualified names. Types may have been retrieved from a
+        # torch.package, which means its typename will be mangled, e.g.
+        # '<torch_package_0>.foo.bar'. Demangle it, and additionally detect any
+        # potential collisions, like with an already existing 'foo.bar'. See
+        # torch/package/mangling.md for more information.
+
+        # demangled name -> original name
+        # This is to detect collisions.
+        demangled_names: Dict[str, str] = {}
+
+        def demangle(name):
```

Comment:
Managing leaking into fx indicates something is not right here. Mangling shouldn't leak at all. Consider using the same approach for any other package code being re-packaged: the exporter needs to know about the importers symbols to successfully package it.